examples/main/main.cpp

```diff
@@ -431,7 +431,7 @@ int main(int argc, char ** argv) {
         }
 
         // end of text token
-        if (embd.back() == llama_token_eos()) {
+        if (embd.empty() || embd.back() == llama_token_eos()) {
```

Comment:
Shouldn't this be `!embd.empty() &&` ..? What is the purpose of hitting this path when `embd` is empty?